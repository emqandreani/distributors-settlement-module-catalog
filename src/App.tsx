import React from "react";

import { AccountInfo } from "./@types/global";
import { AccountContext } from "./context/Account";
import Routes from "./routes";

interface IAppProps {
  /**
   * Account from MSAL
   */
  account?: AccountInfo;
}

const DEFAULT_ACCOUNT: AccountInfo = {
  name: "Wick John",
  username: "thekiller",
  environment: "",
  homeAccountId: "",
  tenantId: "",
  nativeAccountId: "",
  localAccountId: "",
  idTokenClaims: {},
};

const App = ({ account = DEFAULT_ACCOUNT }: IAppProps) => (
  <AccountContext.Provider value={account}>
    <React.Suspense fallback={"...loading"}>
      <Routes />
    </React.Suspense>
  </AccountContext.Provider>
);

export default App;
