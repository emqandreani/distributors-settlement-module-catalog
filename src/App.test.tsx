import { screen } from "@testing-library/react";

import { renderWithProviders } from "./__test__/utils";
import App from "./App";

describe("<App/>", () => {
  test("Should render without crash", () => {
    const { asFragment } = renderWithProviders(<App />);

    expect(asFragment()).toMatchSnapshot();
  });
  test("Should render title", async () => {
    renderWithProviders(<App />);
    const title = await screen.findByText(/Home Remote Page/i);

    expect(title).toBeInTheDocument();
  });
  test("Should render default account", async () => {
    renderWithProviders(<App />);
    const nombre = await screen.findByText(/Nombre: Wick John/i);

    expect(nombre).toBeInTheDocument();
  });
});
