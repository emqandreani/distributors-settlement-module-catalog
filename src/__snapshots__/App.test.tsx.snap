// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<App/> Should render without crash 1`] = `
<DocumentFragment>
  ...loading
</DocumentFragment>
`;
